api/routes/units.js

```diff
@@ -0,0 +1,159 @@
+
+router.get('/', async (req, res, next) => {
+    const status = 200;
+    try {
+        const response = await Unit.find().select('kind')
+        res.json({ status, response })
+        } 
+    catch (error) {
+        console.log(error);
+        const e = new Error("Problem with showing kind for unit.");
+        e.status = 400;
+        next(e);
+    }
+});
+
+
+// GET companies
+router.get(':id/company',async (req, res, next) => {
+    const status = 200
+    const response = await Unit.findById(req.params.id).select('company')
+    res.json({ status, response })
+})
+    // Company.find().then(response => {
+    //     console.log(response)
+    //     res.json({ status, response })
+    // })
+// })
+
+// CREATE
+router.post('/', (req, res, next) => {
+    const status = 201
+    console.log('req.body', req.body)
+    Unit.create(req.body).then(response => {
+        res.status(status).json({ status, response })
+    }).catch(error => {
+        console.error(error)
+        const err = new Error('Problem with creating new post')
+        err.status = 400
+        next(err)
+    })
+})
+
+// GET /api/units?kind=[kind]
+// GET /api/units?floor=[integer]
+// GET /api/v1/units?occupied=[true/false]
+
+// PATCH /api/units/[id] 
+router.patch("/:id", async (req, res, next) => {
+    const status = 201;
+    try{
+        const response = await Unit.findOneAndUpdate({_id: req.params.id}, req.body, {new: true})
+        res.json({status, response})
+    }catch(error){
+        const e = new Error("Unit doesn't exist")
+        e.status = 404
+        next(e)
+    }
+})
+
+
+// PATCH /api/units/[id]/company
+// e.g. PATCH http://localhost:5000/api/v1/units/5/company
+
+router.patch('/:id/company',async (req, res, next) => {
+    const status = 201;
+    try{
+        const response = await Unit.findOneAndUpdate({_id: req.params.id},  {company: {...req.body}}, {new: true})
+        res.json({status, response})
+    }catch(error){
+        const e = new Error("Unit doesn't exist")
+        e.status = 404
+        next(e)
+    }
+})
+
+// DELETE /api/v1/units/[id]/company
+// GET /api/v1/units/[id]/company/employees
+// GET /api/v1/units/[id]/company/employees/[id]
+
+
+
+// POST /api/v1/units/[id]/company/employees
+
+// GET /api/v1/units/[id]
+router.get("/:id", async (req, res, next) => {
+    const status = 200;
+    const response = await Unit.findById(req.params.id);
+    res.json({ status, response });
+});
+
+//DELETE
+router.delete("/:id", async (req, res, next) => {
+    const status = 200;
+    const response = await Unit.findOneAndDelete({ _id: req.params.id });
+    res.json({ status, response });
+});
+
+
+
+
+// PATCH /api/units/[id]/company
+// e.g. PATCH http://localhost:5000/api/v1/units/5/company
+
+router.patch('/:id/companies', (req, res, next) => {
+    const status = 200
+    Company.find().then(response => {
```

Comment:
This is also probably not working, due to the same reason that you're calling `.find()` on a schema.